admin-dev/themes/new-theme/js/components/contextual-notification.ts

```diff
@@ -0,0 +1,119 @@
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import ComponentsMap from '@components/components-map';
+
+/**
+ * This class is responsible for initiating, setting and getting data related to contextual notifications,
+ * that is to say: should we display the notification related to this key identifier in local storage.
+ * It also displays the notification itself
+ */
+
+class ContextualNotification {
+  // all contextual notification data will be stored under this key in local storage
+  private localStorageKey = 'contextual_notifications';
+
+  constructor() {
+    $(document).on(
+      'click',
+      ComponentsMap.contextualNotification.close,
+      (Event) => this.disableNotification(Event),
+    );
+  }
+
+  setItem(key: any, value: boolean): void {
+    const notificationList = JSON.parse(this.getNotificationList());
+    notificationList[key] = value;
+
+    localStorage.setItem(this.localStorageKey, JSON.stringify(notificationList));
+  }
+
+  getItem(key: any): boolean|null {
+    const notificationList = JSON.parse(this.getNotificationList());
+
+    if (key in notificationList) {
+      return notificationList[key];
+    }
+
+    return null;
+  }
+
+  disableNotification(event: any): void {
+    const notificationKey = $(event.target).parent().attr('data-notification-key');
+
+    if (notificationKey !== '') {
+      this.setItem(notificationKey, false);
+    }
+  }
+
+  displayNotification(message: string, key: string): void {
+    const $element = document.createElement('div');
+    $element.classList.add('alert', 'alert-info', 'contextual-notification');
+    $element.setAttribute('data-notification-key', key);
+    $element.innerHTML = `${message}<button type="button" class="close" data-dismiss="alert">&times;</button>`;
+
+    const contentMessageBox = document.getElementById(ComponentsMap.contextualNotification.messageBox);
+
+    if (contentMessageBox instanceof HTMLElement) {
+      contentMessageBox.append($element);
+    }
+  }
+
+  private getNotificationList(): string {
+    return localStorage.getItem(this.localStorageKey) ?? '{}';
+  }
+}
+
+export default function initContextualNotification(key: string): void {
```

Comment:
I would add some JSDoc, I need to reverse engineer every parameter in order to understand it

Reply:
:+1:

Reply:
I agree since it's actually the only public API used it deserves a little bit more explanation Not just the type of the parameter but rather What does this function actually initiates? and a small code example.

Reply:
:+1: